concordium-std/src/test_infrastructure.rs

```diff
@@ -695,6 +655,259 @@ impl<T: AsRef<[u8]>> Seek for ContractStateTest<T> {
     }
 }
 
+impl HasCallResponse for Cursor<Vec<u8>> {
+    fn size(&self) -> u32 { self.data.len() as u32 }
+}
+
+/// Holds a function used for mocking invocations of contracts with
+/// `invoke_contract`.
+pub struct MockFn<State>(
+    Box<dyn Fn(Parameter, Amount, &mut State, &mut Vec<u8>) -> InvokeResult<bool>>,
```

Comment:
Due to re-entrancy the balance of the contract can also be affected by this out-call. So there should probably be a way to allow users to specify that.

Reply:
For example by adding another `&mut Amount` parameter.

Reply:
Another thing,  failure can also return values, but this is not currently possible with this API.

Reply:
I have addressed the latter now.